src/array/ops/cast.rs

```diff
@@ -215,15 +212,38 @@ fn append_values_from_numpy<
     with_match_numeric_daft_types!(datatype, |$N| {
         type Src = <$N as DaftNumericType>::Native;
         let pyarray: PyReadonlyArrayDyn<'_, Src> = pyarray.extract()?;
-        if pyarray.ndim() != 1 {
+        // All of the conditions given in the ndarray::ArrayView::from_shape_ptr() docstring must be met in order to do
+        // the pyarray.as_array conversion: https://docs.rs/ndarray/0.15.6/ndarray/type.ArrayView.html#method.from_shape_ptr
+        // Relevant:
+        //  1. Must be C-contiguous.
+        //  2. Must have non-negative strides.
+
+        // TODO(Clark): Double-check that we're covering all bases here for this unsafe handoff.
+        // TODO(Clark): Relax the C-contiguous constraint, using the ndarray crate to copy the array so it's C-contiguous.
+        // TODO(Clark): Delegate to NumPy's np.ravel() in Python land for the necessary conversions if the ndarray isn't already
+        // C-contiguous and the ndarray crate doesn't expose the requisite functionality, since having to acquire the GIL and
+        // make a copy is better than failing.
+        if !pyarray.is_c_contiguous() {
+            if pyarray.is_fortran_contiguous() {
+                return Err(DaftError::ValueError(format!(
+                    "we only support c-contiguous numpy arrays, got fortran-contiguous ndarray at index: {}",
+                    index
+                )));
+            }
             return Err(DaftError::ValueError(format!(
-                "we only support 1 dim numpy arrays, got {} at index: {}",
-                pyarray.ndim(), index
+                "we only support c-contiguous numpy arrays, but got non-contiguous ndarray at index: {}",
+                index
+            )));
+        }
+        if pyarray.strides().iter().any(|s| *s < 0) {
+            return Err(DaftError::ValueError(format!(
+                "we only support numpy arrays with non-negative strides, but got {:?} at index: {}",
+                pyarray.strides(), index
             )));
         }
 
-        let pyarray = pyarray.downcast::<PyArray1<Src>>().expect("downcasted to numpy array");
-        let sl: &[Src] = unsafe { pyarray.as_slice()}.expect("convert numpy array to slice");
+        let pyarray = pyarray.as_array();
+        let sl: &[Src] = pyarray.as_slice().expect("convert numpy array to slice");
```

Comment:
@samster25 This `pyarray.as_array()` and `ArrayView::as_slice()` is a bit more fast-and-loose then the explicit `PyArray1` downcast, so we might want to harden this a bit more.

Reply:
so does this yield a 1D view of the array?

Reply:
Yep!